node/pkg/common/scissors.go

```diff
@@ -76,3 +76,50 @@ func WrapWithScissors(runnable supervisor.Runnable, name string) supervisor.Runn
 		return runnable(ctx)
 	}
 }
+
+// StartRunnable starts a go routine with the ability to recover from errors by publishing them to an error channel. If catchPanics is true,
+// it will also catch panics and publish the panic message to the error channel. If catchPanics is false, the panic will be propagated upward.
+func StartRunnable(ctx context.Context, errC chan error, catchPanics bool, name string, runnable supervisor.Runnable) {
+	ScissorsErrorsCaught.WithLabelValues(name).Add(0)
+	if catchPanics {
+		ScissorsPanicsCaught.WithLabelValues(name).Add(0)
+	}
+	go func() {
+		if catchPanics {
+			defer func() {
+				if r := recover(); r != nil {
+					var err error
+					switch x := r.(type) {
+					case error:
+						err = fmt.Errorf("%s: %w", name, x)
+					default:
+						err = fmt.Errorf("%s: %v", name, x)
+					}
+					// We don't want this to hang if the listener has already gone away.
+					select {
+					case errC <- err:
+					default:
+					}
+					ScissorsPanicsCaught.WithLabelValues(name).Inc()
+
+				}
+			}()
+		}
+		startRunnable(ctx, errC, name, runnable)
+	}()
+}
+
+// startRunnable is used by StartRunnable. It is a separate function so we can call it directly from tests.
+func startRunnable(ctx context.Context, errC chan error, name string, runnable supervisor.Runnable) {
```

Comment:
can line 48 use this function?

Reply:
I made `RunWithScissors` call `StartRunnable`.